homeassistant/components/homekit/config_flow.py

```diff
@@ -306,24 +308,87 @@ async def async_step_exclude(self, user_input=None):
         )
         return self.async_show_form(step_id="exclude", data_schema=data_schema)
 
+    async def async_step_include(self, user_input=None):
```

Comment:
There is a lot of duplicated code in here between `async_step_include` `async_step_exclude`
Please refactor to break out the duplicate code and make this more DRY

Reply:
I'm not sure what you mean "more DRY"?

At this point I'd argue that breaking this code out into more code will make it less readable.

Reply:
https://en.wikipedia.org/wiki/Don%27t_repeat_yourself